### PR TITLE
[no squash] CI whitespace checks

### PR DESCRIPTION
**Pull request by cx384**
This PR adds a CI workflow to check for trailing whitespaces in the most affected files, and for tabs in the Lua API files. (Other markdown files are not checked.)

I noticed that we repeatedly had problems with whitespaces, so I hope this makes the reviewing of PRs a little bit easier.
The checks are not exhaustive, but most common things should be caught. Other filetypes/checks could be added later if deemed necessary.

(I'm not sure if clang tidy could check for trailing whitespaces, but other files need to be checked too anyway.)

## To do

This PR is Ready for Review.

## How to test
See that trailing whitespaces, and tabs in Lua API files got removed. (They are usually invisible.)

Execute the commands of the workflow on current master, to see how it fails, or check the CI pipeline directly.
``` bash
if git ls-files | grep -E '\.txt$|\.md$|\.[ch]$|\.cpp$|\.hpp$|\.sh$|\.cmake$|\.glsl$' | xargs grep -n '\s$'; then echo -e "\033[0;31mFound trailing whitespace"; (exit 1); else (exit 0); fi
```
``` bash
if grep -n $'\t' doc/lua_api.md doc/client_lua_api.md; then echo -e "\033[0;31mFound tab in markdown file"; (exit 1); else (exit 0); fi
```

### CODE REVIEW

**Comment by sfence** [comment]
@cx384 
Can you add a check like `grep -n -P "^\t*[ ]" tst` to find lines where spaces are used as intents?

File: .github/workflows/whitespace_checks.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: whitespace_checks
+
+# Check whitespaces of the following file types
+# Not checked: .lua, .yml, .properties, .conf, .java, .py, .svg, .gradle, .xml, ...
+# (luacheck already checks .lua files)
```
Comment (appgurueu) [inline .github/workflows/whitespace_checks.yml:5]: Note that Luachecks only detects inconsistent indentation (space followed by tab). So luacheck won't warn about e.g. one line using tabs, another one using spaces.

File: lib/gmp/mini-gmp.c [hunk 2]
```diff
@@ -55,7 +55,7 @@ see https://www.gnu.org/licenses/.  */
 #include <float.h>
 #endif
 
-
```
Comment (appgurueu) [inline lib/gmp/mini-gmp.c:58]: What is going on here?

File: .github/workflows/whitespace_checks.yml (hunk 1 above)
Comment (cx384) [reply .github/workflows/whitespace_checks.yml:5]: This is meant for trailing whitespaces, currently indentation aren't checked by the script anyway.

File: lib/gmp/mini-gmp.c (hunk 2 above)
Comment (cx384) [reply lib/gmp/mini-gmp.c:58]: That's a removal of a form feed character. My script considers it a trailing whitespace.
Just like any other trailing whitespace (e.g. a tab), you usually can't see it, which in my opinion is bad.
I could adjust the script to allow it, but you probably don't want to see any weird character in code files.

Comment (appgurueu) [reply lib/gmp/mini-gmp.c:58]: First time I (don't) see a form feed in source code :sweat_smile:

**Comment by cx384** [comment]
> Can you add a check like `grep -n -P "^\t*[ ]" tst` to find lines where spaces are used as intents?

I already thought about this, but it wouldn't work well, since the current intents aren't very consistent. (And not all file types use tabs for intents.)
It would probably be better to let a C++ format checker do it and simultaneously check for other things.

However, I tried to come up with something, would it be sufficient?
(lib/irrlichtmt/ is not checked, since it frequently uses spaces instead of tabs.)
```
git ls-files | grep -E '^src/.*\.([ch]$|\.cpp$|\.hpp$)' | xargs grep -n '^\s*    '
```
It still has some problems, e.g. in https://github.com/minetest/minetest/blob/d3ca269c79aec18a8c6d938b13af5896ed978136/src/server/clientiface.h#L47-L57)

**Comment by sfence** [comment]
Hmmm.... Understand.

Let's merge this as it is and I will try to add an indent check in a separate PR.

**Review by sfence — APPROVED** [review]
Regex looks fine.

**Pull request merged**